src/Modules/Data/Commands.Data/AddPowerBIDataset.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using Microsoft.PowerBI.Common.Abstractions;
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Api.Datasets;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Data
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = DatasetParameterSetName)]
+    [OutputType(typeof(Dataset))]
+    public class AddPowerBIDataset : PowerBIClientCmdlet, IUserScope
+    {
+        public const string CmdletVerb = VerbsCommon.Add;
+        public const string CmdletName = "PowerBIDataset";
+
+        #region ParameterSets
+        private const string DatasetParameterSetName = "Dataset";
+        private const string WorkspaceParameterSetName = "Workspace";
+        #endregion
+
+        #region Parameters
+        [Parameter(Mandatory = true)]
+        public Dataset Dataset { get; set; }
+       
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false, ParameterSetName = DatasetParameterSetName)]
```

Comment:
I think you should make another parameter set for workspace ID and move the DatasetParameterSetName attribute to the Dataset parameter.

Reply:
I originally added four parameter sets for datasetid, dataset, workplace, workplaceid but couldn't figure out how to set them as parametersetname to work as I want. Do you have any good sample for it?

Reply:
@CodeCyclone Can you provide a good example for Kenichiro to follow here?

Reply:
For now, I will commit the current update without this.

Reply:
I'm good with this design